web/src/app/admin/admin.api.service.ts

```diff
@@ -12,7 +12,9 @@ export class AdminApiService {
   constructor(private httpClient: HttpClient) {}
 
   public async getAllProducts(): Promise<Product[]> {
-    const result = await this.httpClient.get(adminRoute + '/products').toPromise();
+    const result = await this.httpClient
+      .get(adminRoute + '/products')
```

Comment:
use string interpolation.
`${adminRoute}/products`

Reply:
Working on it